CMakeLists.txt

```diff
@@ -579,63 +575,16 @@ if (WIN32)
-
-        # Warnings that are thrown on standard libraries and not OpenMW
-        4347 # Non-template function with same name and parameter count as template function
-        4365 # Variable signed/unsigned mismatch
-        4510 4512 # Unable to generate copy constructor/assignment operator as it's not public in the base
-        4706 # Assignment in conditional expression
-        4738 # Storing 32-bit float result in memory, possible loss of performance
-        4774 # Format string expected in argument is not a string literal
-        4986 # Undocumented warning that occurs in the crtdbg.h file
-        4987 # nonstandard extension used (triggered by setjmp.h)
-        4996 # Function was declared deprecated
-
-        # caused by OSG
-        4589 # Constructor of abstract class 'osg::Operation' ignores initializer for virtual base class 'osg::Referenced' (False warning)
-
-        # caused by boost
-        4191 # 'type cast' : unsafe conversion (1.56, thread_primitives.hpp, normally off)
-        4643 # Forward declaring 'X' in namespace std is not permitted by the C++ Standard. (in *_std_fwd.h files)
-        5204 # Class has virtual functions, but its trivial destructor is not virtual
-
-        # caused by MyGUI
-        4297 # function assumed not to throw an exception but does
-
         # OpenMW specific warnings
-        4099 # Type mismatch, declared class or struct is defined with other type
         4100 # Unreferenced formal parameter (-Wunused-parameter)
-        4101 # Unreferenced local variable (-Wunused-variable)
         4127 # Conditional expression is constant
-        4242 # Storing value in a variable of a smaller type, possible loss of data
         4244 # Storing value of one type in variable of another (size_t in int, for example)
-        4245 # Signed/unsigned mismatch
         4267 # Conversion from 'size_t' to 'int', possible loss of data
-        4305 # Truncating value (double to float, for example)
-        4309 # Variable overflow, trying to store 128 in a signed char for example
-        4351 # New behavior: elements of array 'array' will be default initialized (desired behavior)
-        4355 # Using 'this' in member initialization list
-        4464 # relative include path contains '..'
-        4505 # Unreferenced local function has been removed
-        4701 # Potentially uninitialized local variable used
-        4702 # Unreachable code
-        4714 # function 'QString QString::trimmed(void) &&' marked as __forceinline not inlined
-        4800 # Boolean optimization warning, e.g. myBool = (myInt != 0) instead of myBool = myInt
-        )
-
-    if (MSVC_VERSION GREATER 1800)
-        set(WARNINGS_DISABLE ${WARNINGS_DISABLE} 5026 5027
-            5031 # #pragma warning(pop): likely mismatch, popping warning state pushed in different file (config_begin.hpp, config_end.hpp)
+        4996 # Function was declared deprecated
```

Comment:
If we aren't warned about deprecated functions, we get much less notice and therefore time to migrate. I'm not sure this one's a good idea. Actually, I'm not sure it's sensible to have the possible-loss-of-data warning above disabled, either.

Reply:
We already have these warnings disabled in master for ten years.

I'm fine to enable 4996 if you think that it is needed, but probably it would be better to create a task on bugtracker instead and specify functions which should be replaced. So far I do not know how to replace them in a cross-platform way.

4267 will require a plenty of changes in the OpenMW codebase (there is at least 67 warnings), so better to fix it in the follow-up merge request. Also in many cases such warning comes from Qt (where "getCount"-style functions return int instead of size_t), so a "fix" will be just a `static_cast<int>` in such cases).

Reply:
https://gitlab.com/OpenMW/openmw/-/issues/5994

Reply:
We can create an issue for 4267 after merging this - usually it won't matter, but there might be the odd case of overflow. Coverity *might* be detecting those for us, so this might be fuss about nothing, but if it turns out it isn't, it's safer to have everything manually labelled as *a human thought about this and decided it was safe* with an explicit cast.